src/distribution_finction.py

```diff
@@ -0,0 +1,235 @@
+from os import makedirs
+import numpy as np
+import plotly.graph_objects as go
+from sympy import Symbol, pi, E, sqrt, integrate, oo
+from abc import ABC, abstractmethod
+from scipy import stats
+import math
+from config import FIGS_ROOT
+
+
+class ProbabilisticDistribution(ABC):
+
+    @abstractmethod
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def show_figure(self):
+        pass
+
+    @abstractmethod
+    def save_figure(self):
+        pass
+
+
+class Visualizer:
+
+    @staticmethod
+    def visualize(x, y, title, fig_processing) -> str:
+        '''Show or save figure.
+
+        Args:
+            x: np.ndarray
+                data points
+            y: np.ndarray
+                value of probability
+            title: str
+                figure title
+            fig_processing: str
+                show or save
+        '''
+        fig = go.Figure()
+        fig.add_trace(go.Scatter(x=x, y=y, name=title))
+        fig.update_xaxes(title='x')
+        fig.update_yaxes(title='y')
+        fig.update_layout(title=title)
+        if fig_processing == 'show':
+            fig.show()
```

Comment:
return None
elif fig_processing == 'save': -> if fig_processing == 'save':

Reply:
If fig_processing is 'show', the return type is None.
If fig_processing is 'save', the return type is str.
What should be the return type of the visualize function?